### PR TITLE
Add "Portuguese (Brazil)" for audio and subtitles streams (ISO-639-2)

### PR DESCRIPTION
**Pull request by Wanilton**
## Description
This adds the "Portuguese (Brazil)" language code to the language code expander, so this would show up in the gui instead of "unknown". Matrix

## Motivation and Context
Used in the latest 2 years here, work fine.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ X ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [X] All new and existing tests passed

### CODE REVIEW

**Comment by phunkyfish** [comment]
Please don't open a new PR as you lost all the comments, history and context  other users have provided.

Here is the previous PR: https://github.com/xbmc/xbmc/pull/17665

It would be best to close this PR and let someone help you fix the previous one.

**Comment by Wanilton** [comment]
The previous have one error in the code, when I try fix, I create one second commit, sorry.

The code is fine now, only need include milestone and label to pass.

**Comment by phunkyfish** [comment]
Yes, because you can't fix it via the web interface. You need to fix it on the command line.

**Comment by phunkyfish** [comment]
@scott967 here is your comment which is still valid on what I think is the 3 version of this PR.

"I don't think there are any such 639-1 or 639-2 codes? I understand the reason for this, it seems there needs to be a policy with respect to adding non-spec items to lists the purport to represent an official standard."

@Wanilton please do not open any new PRs for this change. It makes it impossible to follow what is going on.

**Comment by phunkyfish** [comment]
jenkins build this please

**Review by phunkyfish — CHANGES_REQUESTED** [review]
How this is supposed to work is a two letter lower case code for the language and a two letter uppercase code for the country.

So for this example: `pt-BR`.

So there is no Portuguese Brasil language code.

Can you please give examples of what is detected in the streams?

This may work but it might not be the correct way to do it.

File: xbmc/utils/LangCodeExpander.cpp [hunk 1]
```diff
@@ -1652,6 +1655,7 @@ const std::array<ISO3166_1, 245> RegionCodes = {{
     {"pw", "plw"},
     {"ps", "pse"},
     {"pa", "pan"},
+    {"pb", "pob"},
```
[the remark this answers is not in the document]
Comment (MediaBrasil) [reply xbmc/utils/LangCodeExpander.cpp:1658]: @pkunkyfish, Portuguese is one language only, but its similar american enghish and american british, have distinct words, and services like Netflix have Portuguese and Portuguese (brazil), addon Netflix use a hack to work in Kodi. When I exclude hack Netflix and included support in Kodi, all work fine to me. I never have issue using the code. The online database used by scrapers has 2 languages to Portuguese and Portuguese (Brazil) too. Subtitles services have Portuguese and Portuguese (Brazil). GUI interface in Kodi is the same, has 2 too. Only don´t have to audio and subtitle, don´t make any sense don´t have. Get unknow is bad for us.

Here 2 languages in themoviedb

https://i.imgur.com/0e6XdVR.jpg

https://emby.media/community/index.php?/topic/9013-please-add-opensubtitles-brazilian-portuguese-language-option-for-subtitles-download/

https://www.filebot.net/forums/viewtopic.php?t=2543

[the remark this answers is not in the document]
Comment (phunkyfish) [reply xbmc/utils/LangCodeExpander.cpp:1658]: I'm not disagreeing that `"pb, pob"` is an unofficial working language code.

But the array in this comment is for Region codes, not language codes. A language code has no place in this array, official or not.

[the remark this answers is not in the document]
Comment (howie-f) [reply xbmc/utils/LangCodeExpander.cpp:1658]: @Wanilton, to shorten things a little i offer my help. You can invite me as a collaborator to your repository under "Settings" => "Manage access". I would then fix this thing for you. Please do not delete your fork, thanks

[the remark this answers is not in the document]
Comment (Wanilton) [reply xbmc/utils/LangCodeExpander.cpp:1658]: Done, thanks for your help. Invite sent.

[the remark this answers is not in the document]
Comment (phunkyfish) [reply xbmc/utils/LangCodeExpander.cpp:1658]: Thanks @howie-f

@Wanilton @MediaBrasil if you test this PR but don’t change the Region codes array does it still work?

[the remark this answers is not in the document]
Comment (Wanilton) [reply xbmc/utils/LangCodeExpander.cpp:1658]: I need to compile and test if it continues ok. I will try later here.

[the remark this answers is not in the document]
Comment (phunkyfish) [reply xbmc/utils/LangCodeExpander.cpp:1658]: That would be great. Thanks

[the remark this answers is not in the document]
Comment (howie-f) [reply xbmc/utils/LangCodeExpander.cpp:1658]: yes ok

**Comment by phunkyfish** [comment]
@howie-f 

It would be good to leave a comment before the inserted lines to make it clear that it’s an unofficial entry.

**Comment by howie-f** [comment]
ups, sorry for that 😳

**Comment by phunkyfish** [comment]
> ups, sorry for that 😳
> now waiting for response from @Wanilton before i push

No worries. Thanks again for helping!

**Comment by Wanilton** [comment]
I removed line 1658, and recompile, continue working ok. Sorry, it´s my first PR, and when to update the Kodi code create a second commit.

**Comment by howie-f** [comment]
ok, thanks for testing. now we have to wait if the build runs ok. please leave the branch as is ;)

File: xbmc/utils/LangCodeExpander.cpp [hunk 2]
```diff
@@ -680,6 +680,7 @@ const std::array<struct LCENTRY, 185> g_iso639_1 = {{
     {MAKECODE('\0', '\0', 'p', 'l'), "Polish"},
     {MAKECODE('\0', '\0', 'p', 's'), "Pashto, Pushto"},
     {MAKECODE('\0', '\0', 'p', 't'), "Portuguese"},
+    {MAKECODE('\0', '\0', 'p', 'b'), "Portuguese (Brazil)"},
```
Comment (phunkyfish) [inline xbmc/utils/LangCodeExpander.cpp:683]: Can we add a comment before this line. Something like:

`// unofficial language code for Brazilian Portuguese`

File: xbmc/utils/LangCodeExpander.cpp [hunk 3]
```diff
@@ -1135,6 +1136,7 @@ const std::array<struct LCENTRY, 539> g_iso639_2 = {{
     {MAKECODE('\0', 'p', 'o', 'n'), "Pohnpeian"},
     {MAKECODE('\0', 'p', 'o', 'l'), "Polish"},
     {MAKECODE('\0', 'p', 'o', 'r'), "Portuguese"},
+    {MAKECODE('\0', 'p', 'o', 'b'), "Portuguese (Brazil)"},
```
Comment (phunkyfish) [inline xbmc/utils/LangCodeExpander.cpp:1139]: Same here

File: xbmc/utils/LangCodeExpander.cpp [hunk 4]
```diff
@@ -1413,6 +1415,7 @@ const std::array<ISO639, 189> LanguageCodes = {{
     {"om", "orm", NULL, NULL},
     {"os", "oss", NULL, NULL},
     {"pa", "pan", NULL, NULL},
+    {"pb", "pob", NULL, NULL},
```
Comment (phunkyfish) [inline xbmc/utils/LangCodeExpander.cpp:1418]: And here

**Review by phunkyfish — COMMENTED** [review]
Just some comments to add

File: xbmc/utils/LangCodeExpander.cpp (hunk 2 above)
Comment (howie-f) [reply xbmc/utils/LangCodeExpander.cpp:683]: sure

File: xbmc/utils/LangCodeExpander.cpp (hunk 3 above)
Comment (howie-f) [reply xbmc/utils/LangCodeExpander.cpp:1139]: done

File: xbmc/utils/LangCodeExpander.cpp (hunk 4 above)
Comment (howie-f) [reply xbmc/utils/LangCodeExpander.cpp:1418]: done

**Review by phunkyfish — APPROVED** [review]
Looks good once Jenkins is happy

**Comment by phunkyfish** [comment]
Thanks for the contribution @Wanilton !

**Pull request merged**